src/meili/traffic_segment_matcher.cc

```diff
@@ -65,15 +67,14 @@ namespace {
       auto segments = tile->GetTrafficSegments(edge);
       //merge them into single entries per segment id
       for(const auto& segment : segments) {
-        //new one
-        if(merged.empty() || merged.back()->segment_id_ != segment.segment_id_) {
```

Comment:
this was just wrong. we merge adjecent ots entries into a single one keeping track of the start/end edgeid/percentage. this had the sideeffect that when the trace made a loop, you could merge two segments that were the same id together even though you left the segment on a loop and got back on it a second time. in that case we want two entries and not to merge them which this was doing